### DISCUSSION TITLE
Building progams without needing to specify libraries and source files?

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [mesonbuild/meson#9681] [Q&A] [answered]
Is it possible to add support for building directly from a main file without needing to specify every single file and library used?
Something similar to .net's build system?

Example:

`meson main.c <ProgramName> <ProgramVersion> <Flags>`

**Comment by eli-schwartz** [comment, marked as answer]
meson is not going to parse your main.c and figure out what functions it makes use of. I don't know what the .net build system does...

What I do know is that meson provides functionality to get you most of the way there, by generating a project.

```
$ set -x; meson init *.c --name tinyproj --version 1.2
+ meson init bar.c foo.c --name tinyproj --version 1.2
Using "tinyproj" (project name) as name of executable to build.
Detected language: c
Generated meson.build file:

project('tinyproj', 'c',
  version : '1.2',
  default_options : ['warning_level=3'])

executable('tinyproj',
           'bar.c',
           'foo.c',
           install : true)
$ meson setup builddir/ <Flags>
```
